File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <stack>
using namespace std;

bool solution(string s)
{
    stack<char> st;
    int i = 0;
    while (i < s.length()) {
        st.push(s[i]);
        if (st.size() >= 2) {
            char tmp1, tmp2;
            tmp1 = st.top();
            st.pop();
            tmp2 = st.top();
            st.pop();
            if (!(tmp2 == '(' && tmp1 == ')')) {
                st.push(tmp2);
                st.push(tmp1);
            }
        }
        i++;
    }
    return st.empty();
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    st = []
    idx = 0
    while idx < len(s) :
        st.append(s[idx])
        if len(st) >= 2 :
            tmp1 = st.pop()
            tmp2 = st.pop()
            if (tmp2 == '(' and tmp1 == ')') == False :
                st.append(tmp2)
                st.append(tmp1)
        idx += 1
    if (len(st) == 0) :
        return True
    else :
        return False